pkg/apis/core/validation/shoot_test.go

```diff
@@ -4527,6 +4527,31 @@ var _ = Describe("Shoot Validation Tests", func() {
 				}),
 			)
 		})
+
+		Context("scheduler name", func() {
+			It("allow setting the default scheduler name", func() {
+				shoot.Spec.SchedulerName = nil
+				oldShoot := shoot.DeepCopy()
+				shoot.Spec.SchedulerName = pointer.String("default-scheduler")
+
+				errorList := ValidateShootSpecUpdate(&shoot.Spec, &oldShoot.Spec, metav1.ObjectMeta{}, field.NewPath("spec"))
+
+				Expect(errorList).To(BeEmpty())
+			})
+
+			It("forbid changing the scheduler name", func() {
+				shoot.Spec.SchedulerName = nil
+				oldShoot := shoot.DeepCopy()
+				oldShoot.Spec.SchedulerName = pointer.String("foo-scheduler")
+
+				errorList := ValidateShootSpecUpdate(&shoot.Spec, &oldShoot.Spec, metav1.ObjectMeta{}, field.NewPath("spec"))
+
+				Expect(errorList).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeInvalid),
+					"Field": Equal("spec.schedulerName"),
+				}))))
+			})
+		})
```

Comment:
Maybe add a test for the case where `shoot.Spec.SchedulerName` is not nil and scheduler is changed from value `a` to `b`